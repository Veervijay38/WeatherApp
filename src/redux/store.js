import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import weatherReducer from "./reducer/reducers";

const middleware = getDefaultMiddleware({
  serializableCheck: {
    // Increase the warning threshold (default is 32ms)
    warningTimeout: 50,
  },
});

const store = configureStore({
  reducer: {
    weather: weatherReducer,
  },
  middleware,
});

export default store;
