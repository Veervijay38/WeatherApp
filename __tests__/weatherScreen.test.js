import React from "react";
import { render, waitFor, act } from "@testing-library/react-native";
import { Provider } from "react-redux";
import store from "../src/redux/store";
import WeatherScreen from "../src/screens/weatherScreen";
import { NavigationContainer } from "@react-navigation/native";

it("Weather Component > renders without errors and matches snapshot", () => {
  const { toJSON } = render(
    <Provider store={store}>
      <NavigationContainer>
        <WeatherScreen />
      </NavigationContainer>
    </Provider>
  );

  expect(toJSON()).toMatchSnapshot();
});

// it("displays error message on API error", async () => {
//     act(async () => {
//       const { getByText } = render(
//         <Provider store={store}>
//           <NavigationContainer>
//             <WeatherScreen />
//           </NavigationContainer>
//         </Provider>
//       );

//       await waitFor(() => {
//         expect(getByText("An error occurred")).toBeTruthy();
//       });
//     });
//   });
