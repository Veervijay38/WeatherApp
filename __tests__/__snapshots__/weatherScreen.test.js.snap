// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Component > renders without errors and matches snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      blurRadius={70}
      resizeMode="cover"
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "overflow": "hidden",
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          }
        }
        useNativeDriver={false}
      >
        <RNSVGSvgView
          bbHeight="140"
          bbWidth="140"
          focusable={false}
          height={140}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 140,
                "width": 140,
              },
            ]
          }
          width={140}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGPath
              d="M64.50916208442445 15.274770909708586
          A55 55 0 0 1 70 15"
              fill={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                ]
              }
              stroke={
                {
                  "payload": 4278956978,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeWidth="10"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;
